pkg/trace/writer/trace.go

```diff
@@ -275,6 +275,7 @@ func (w *TraceWriter) serializer() {
 				"Content-Encoding": "gzip",
 				headerLanguages:    strings.Join(info.Languages(), "|"),
 			})
+			p.body.Grow(len(b) / 2)
```

Comment:
Any reason you picked `/2` specifically? I assume it's because after compression we won't actually need the full `len(b)`?

Reply:
`len(b) / 2` showed better test results compared to `len(b)` which makes sense since the compression reduces the size. Haven't tried other values tbh.